### PR TITLE
[CloudSyncMgr] Renaming the CloudSyncMgr API according to functionality to add more APIs

### PR DESCRIPTION
**Pull request by nitu-s-gupta**
# Description

The APIs were names as per the naming convection followed for previous APIs as for example, APIV1Request<<PackageName>><<GET/POST>> But with multiple APIs being added going further in CloudSyncMgr it is necessary to modify the API in terms of functionality i.e publish, subscribe

Fixes # (#382 )

## Type of change

Please delete options that are not relevant.
- [X] Code cleanup/refactoring

# How Has This Been Tested?
1.MQTT mosquitto broker is configured to be running in the AWS endpoint.
2. The edge orchestration is build and run using following command with option CLOUD_SYNC set to true
```
docker run -it -d --privileged --network="host" --name edge-orchestration -e CLOUD_SYNC=true -v /var/edge-orchestration/:/var/edge-orchestration/:rw -v /var/run/docker.sock:/var/run/docker.sock:rw -v /proc/:/process/:ro  lfedge/edge-home-orchestration-go:latest
```
3. From the another terminal/post make a curl command as follows to publish data using home edge to the broker running on AWS endpoint

```
curl --location --request POST 'http://192.168.1.107:56001/api/v1/orchestration/cloudsyncmgr/publish' \
--header 'Content-Type: text/plain' \
--data-raw '{
    "appid": "testsamsung",
    "payload": "Testdata for subscribing",
    "topic": "homeedge/livingroom",
    "url" : "ec2-184-73-83-74.compute-1.amazonaws.com"
}'
```
4.You should obtain the following message for successful publish
```
{"Message":"Data published successfully to Cloud","RemoteTargetInfo":"","ServiceName":""}
```
5. The Broker logs are checked in AWS and we observe following logs
```
1641480800: New connection from 49.207.219.60 on port 1883.
1641480800: New client connected from 49.207.219.60 as com.nokia.homeedge1 (p2, c1, k30).
1641480800: No will message specified.
1641480800: Sending CONNACK to com.nokia.homeedge1 (0, 0)
1641480800: Received PUBLISH from com.nokia.homeedge1 (d0, q0, r1, m0, 'home1/livingroom', ... (80 bytes))
1641480830: Received PINGREQ from com.nokia.homeedge1
1641480830: Sending PINGRESP to com.nokia.homeedge1
1641480850: Received PUBLISH from com.nokia.homeedge1 (d0, q0, r1, m0, 'home1/livingroom', ... (73 bytes))
```
Same client sending data but connection is established only once.

**Test Configuration**:
* OS type & version: ( Ubuntu 20.04)
* Hardware: (x86-64)
* Edge Orchestration Release: (v1.1.0)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/controller/cloudsyncmgr/cloudsync.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ const (
 type CloudSync interface {
 	InitiateCloudSync(isCloudSet string) error
 	//implemented by external REST API
-	RequestCloudSyncConf(host string, clientID string, message mqttmgr.Message, topic string) string
+	RequestCloudSyncPublish(host string, clientID string, message mqttmgr.Message, topic string) string
```
Comment (t25kim) [inline internal/controller/cloudsyncmgr/cloudsync.go:39]: Simply, what about `Publish`?

Comment (nitu-s-gupta) [reply internal/controller/cloudsyncmgr/cloudsync.go:39]: Do you mean just Publish(), I thought of this idea, but then thought would be confusing for nay neww developer to understand by just Publish and hence this name.

Comment (t25kim) [reply internal/controller/cloudsyncmgr/cloudsync.go:39]: However this function is in the CloudSync. What about `RequestPublish` instead?

Comment (nitu-s-gupta) [reply internal/controller/cloudsyncmgr/cloudsync.go:39]: seems good

**Comment by t25kim** [comment]
Please correct `How Has This Been Tested?`.
I think you should fix the [document](https://github.com/lf-edge/edge-home-orchestration-go/blob/master/docs/cloudsync_mqtt.md#6-run-cloudsync) as well.

**Comment by nitu-s-gupta** [comment]
> Please correct `How Has This Been Tested?`. I think you should fix the [document](https://github.com/lf-edge/edge-home-orchestration-go/blob/master/docs/cloudsync_mqtt.md#6-run-cloudsync) as well.

Done!!!

**Review by suresh-lc — APPROVED** [review]
+1

**Comment by nitu-s-gupta** [comment]
@t25kim @MoonkiHong @tdrozdovsky PTAL 🙏

**Review by tdrozdovsky — CHANGES_REQUESTED** [review]
Please change

**Comment by tdrozdovsky** [comment]
Please change : "p, member, /api/v1/orchestration/cloudsyncmgr/, *\n" to "p, member, /api/v1/orchestration/cloudsyncmgr/publish" line 39: `internal/controller/securemgr/authorizer/authorizer.go`

**Comment by tdrozdovsky** [comment]
Please change: 
`docs/secure_manager.md`:
Line 225: | /api/v1/orchestration/cloudsyncmgr | Allow | Allow  | => | /api/v1/orchestration/cloudsyncmgr/publish | Allow | Allow  |
Line 248: p, member, /api/v1/orchestration/cloudsyncmgr, *  => p, member, /api/v1/orchestration/cloudsyncmgr/publish, *
`internal/controller/securemgr/authorizer/authorizer.go:`
Line 39: "p, member, /api/v1/orchestration/cloudsyncmgr, *\n" => "p, member, /api/v1/orchestration/cloudsyncmgr/publish, *\n

File: internal/orchestrationapi/mocks/mock_orchestration.go [hunk 2]
```diff
@@ -105,17 +105,17 @@ func (mr *MockOrcheExternalAPIMockRecorder) RequestService(arg0 interface{}) *go
...
-func (m *MockOrcheExternalAPI) RequestCloudSync(arg0 string, arg1 string, arg2 mqtt.Message, arg3 string) string {
+func (m *MockOrcheExternalAPI) RequestCloudSyncPublish(arg0 string, arg1 string, arg2 mqtt.Message, arg3 string) string {
 	m.ctrl.T.Helper()
-	ret := m.ctrl.Call(m, "RequestCloudSync", arg0, arg1, arg2, arg3)
+	ret := m.ctrl.Call(m, "RequestCloudSyncPublish", arg0, arg1, arg2, arg3)
 	ret0, _ := ret[0].(string)
 	return ret0
 }
 
 // RequestCloudSync indicates an expected call of RequestService.
```
Comment (tdrozdovsky) [inline internal/orchestrationapi/mocks/mock_orchestration.go:115]:
```suggestion
// RequestCloudSyncPublish indicates an expected call of RequestService.
```

**Review by tdrozdovsky — CHANGES_REQUESTED** [review]

File: internal/orchestrationapi/mocks/mock_orchestration.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nitu-s-gupta) [reply internal/orchestrationapi/mocks/mock_orchestration.go:115]: Done!!

**Comment by nitu-s-gupta** [comment]
> Please change: `docs/secure_manager.md`: Line 225: | /api/v1/orchestration/cloudsyncmgr | Allow | Allow | => | /api/v1/orchestration/cloudsyncmgr/publish | Allow | Allow | Line 248: p, member, /api/v1/orchestration/cloudsyncmgr, * => p, member, /api/v1/orchestration/cloudsyncmgr/publish, * `internal/controller/securemgr/authorizer/authorizer.go:` Line 39: "p, member, /api/v1/orchestration/cloudsyncmgr, *\n" => "p, member, /api/v1/orchestration/cloudsyncmgr/publish, *\n `internal/restinterface/externalhandler/externalhandler_test.go` Line:292: func TestAPIV1RequestCloudSyncmgrPost(t *testing.T) { => func TestAPIV1RequestCloudSyncmgrPublish(t *testing.T) {

Done!!

**Review by tdrozdovsky — APPROVED** [review]
+1

**Review by t25kim — APPROVED** [review]
LGTM Thanks!

**Review by MoonkiHong — APPROVED** [review]
LGTM.

**Pull request merged**